src/main/java/org/jenkinsci/plugins/workflow/log/FileLogStorage.java

```diff
@@ -126,7 +134,7 @@ private void checkId(String id) throws IOException {
         assert Thread.holdsLock(this);
         if (!Objects.equals(id, lastId)) {
             bos.flush();
-            long pos = os.getChannel().position();
+            long pos = osStartPosition + cos.getByteCount();
```

Comment:
SpotBugs complaints

Reply:
Any idea offhand what the cost of reentrant synchronization is on a method call? We could delete `assert Thread.holdsLock(this);` in `checkId` and just make the method `synchronized` to help SpotBugs understand what is happening, but maybe that has a nontrivial cost?

Reply:
https://github.com/jenkinsci/workflow-api-plugin/pull/296/commits/b763772ecd87592dbfe8b1bf0a5bb66cde79393f suppresses the warnings though.